### PR TITLE
Open links in new pages, refactor

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR solves the issue raised here: https://github.com/gnosis/cowswap/pull/1200#issuecomment-895090065

Additionally it adds analytics tracking also for Markdown links. 

## Implementation details
For fixing these two things I refactored the code that rendered the links for the content pages. 

The behaviour was similar, but it didn't need the scroll logic. So I did a refactor where now we have 3 link components:
* `ExternalLink`: It's a link that open a URL in a new tab and tracks it in analytics
* `Link`: It's a component that decides depending on the path if it's an `ExternalLink` or if it's `HashLink` (internal one).
* `LinkScrollable`: It's what it is used for the content pages. It's a `Link` and has the scroll logics. Mantains the previous logic, it was just renamed and refactored.

Additionally, cleans up all dependencies to `react-markdown` so we only depend to it in `components/Markdown`
Now this file exports:
* `MarkdownPage`:  Before it was called `Markdown`. It uses `MarkdownContent` for rendering its links.
* `Markdown`: Basic markdown with the smart link logic that would add analytics automatically

## Test 
1. Test all content pages: FAQ, About, Terms and Conditions, so links still work
2. Verify links work and open in a new tab

### CODE REVIEW

File: src/custom/components/Link/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    </HashLink>
+  )
+}
+
+export function LinkScrollable(props: LinkRendererProps): JSX.Element {
+  const { children, smooth = true, ...otherProps } = props
+
+  const scrollWithOffset = (el: HTMLElement) => {
+    const yCoordinate = el.getBoundingClientRect().top + window.pageYOffset
+    const yOffset = -24
```
Comment (alfetopito) [inline src/custom/components/Link/index.tsx:31]: :sparkles:

File: src/custom/components/Markdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {file && <MarkdownContent>{file}</MarkdownContent>}
+        {error && <MarkdownContent>{error}</MarkdownContent>}
+      </Content>
+      {/* <ScrollToTop
+        styleProps={{
+          bottom: '8.8%',
+          right: 'calc(50% - 4.6rem)',
+          background: '#9bd7c2'
+        }}
+      /> */}
```
Comment (alfetopito) [inline src/custom/components/Markdown/index.tsx:42]: Any reason to keep?

**Review by alfetopito — APPROVED** [review]
Looks good, only minor comments

**Comment by elena-zh** [comment]
Changes LGTM. 
However, I have found 2 links in the app that do not open their content in another tab: 
1. FAQ link at the bottom of the 'About' page
not new tab

2. FAQ link in the 'Unsupported Token' modal
unsupp token

File: src/custom/components/Link/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/components/Link/index.tsx:31]: this mean magic number :D

File: src/custom/components/Markdown/index.tsx (hunk 2 above)
Comment (W3stside) [reply src/custom/components/Markdown/index.tsx:42]: yes, i made it therefore it should stay forever

**Review by W3stside — APPROVED** [review]

File: src/custom/components/Link/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/custom/components/Link/index.tsx:31]: Why I get the blame  of some code i didn't even do 😂. I don't even know what's that :) 

This is just a refactor of some code that someone in the team did some time ago. Not giving names, only mentioning that he is blonde

Comment (anxolin) [reply src/custom/components/Link/index.tsx:31]: Let's review this, but for DEVELOP. Okok?

File: src/custom/components/Markdown/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/custom/components/Markdown/index.tsx:42]: Okok, I will upload it to the blockchain and IPFS and then im deleting it. :P

Comment (anxolin) [reply src/custom/components/Markdown/index.tsx]: I just did this: https://github.com/gnosis/cowswap/pull/1212/commits/bded395259fe7c10a77b7d85aa504a7eb13bf866

Comment (anxolin) [reply src/custom/components/Markdown/index.tsx]: Done: https://github.com/gnosis/cowswap/pull/1212/commits/1025cc93d68daad1f48ac553aada3a9387410019

**Comment by anxolin** [comment]
> Changes LGTM.
However, I have found 2 links in the app that do not open their content in another tab:

This is alright, it's intended. We just do a normal navigation there.

**Pull request merged**